editor/src/components/inspector/sections/layout-section/layout-system-subsection/split-chained-number-input.tsx

```diff
@@ -320,6 +320,12 @@ export const SplitChainedNumberInput = React.memo((props: SplitChainedNumberInpu
 
   const isCmdPressedRef = useRefEditorState((store) => store.editor.keysPressed.cmd === true)
 
+  const selectedViews = useEditorState(
+    Substores.selectedViews,
+    selectedViewsSelector,
+    'PaddingControl selectedViews',
+  )
+
```

Comment:
hmm this is a tougher nut to crack – as far as I understand, if selectedViews change, you want to call `updateMode()` and `setMode(null)` – without a major refactor, i think the most efficient way to do this is to use a `useSelectorWithCallback` – it's the same as a useEditorState, but instead of forcing a re-render, it merely executes a callback function.

pseudo code:
```typescript
 useSelectorWithCallback(
    Substores.selectedViews,
    selectedViewsSelector,
    () => {
      updateMode();
      setMode(null);
    },
    'PaddingControl selectedViews',
  )
```

Reply:
During discussions we decided against implementing this suggestion in this PR